tests/testthat/test-datatype.R

```diff
@@ -42,42 +42,32 @@ test_that("POSIXct data conversion", {
     pl$lit("2022-01-01")$str$strptime(pl$Datetime(), "%F")$to_r(),
     as.POSIXct("2022-01-01")
   )
-  # TODO: infer timezone from string, change the arugment name from `tz`
+
+  # Note: do not set the "TZ" envvar here as it doesn't work consistently
+  # https://github.com/pola-rs/r-polars/issues/1188
```

Comment:
Under what conditions does this not work? I suspect this is a function bug, not a testing issue.

Reply:
I think #1188 should have proved that envvar is not reliable.

Reply:
I mean to say that there may be a bug in polars that does not respect envvar.

Reply:
I don't think envvars are universal. Unless you can reference a standard that explicitly was agreed upon by all operating systems that `TZ` is to be regarded as the source of truth when checking what time zone to operate in. So this is quite irrelevant, `polars` behaves correctly when explicitly told to use a specific time zone. And since these test are time_zone specific it is very sensible to explicitly state the time zone.

Reply:
Worth noting that `Sys.timezone()` uses this envvar (and then has a fallback):

```r
function (location = TRUE) 
{
    if (!location) 
        .Deprecated(msg = "Sys.timezone(location = FALSE) is defunct and ignored")
    if (!is.na(tz <- get0(".sys.timezone", baseenv(), mode = "character", 
        inherits = FALSE, ifnotfound = NA_character_))) 
        return(tz)
    cacheIt <- function(tz) {
        unlockBinding(".sys.timezone", baseenv())
        assign(".sys.timezone", tz, baseenv())
        lockBinding(".sys.timezone", baseenv())
    }
    tz <- Sys.getenv("TZ")
	...
```

However I don't see any `TZ` in the polars source code upstream.